src/Compilers/CSharp/Test/Emit2/PDB/PDBTests.cs

```diff
@@ -367,6 +368,83 @@ public void SymWriterErrors4()
             Assert.False(result.Success);
         }
 
+        [ConditionalTheory(typeof(WindowsOnly), Reason = ConditionalSkipReason.NativePdbRequiresDesktop)]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/75237")]
+        [InlineData(0x9_999)]
+        [InlineData(0x9_000)]
+        public void NativeWriterLimit_Under(int length)
+        {
+            CompileWithMockedCustomMetadata(length).Diagnostics.Verify();
+        }
+
+        [ConditionalTheory(typeof(WindowsOnly), Reason = ConditionalSkipReason.NativePdbRequiresDesktop)]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/75237")]
+        [InlineData(0x10_000)]
+        [InlineData(0x20_000)]
+        public void NativeWriterLimit_Over(int length)
+        {
+            CompileWithMockedCustomMetadata(length).Diagnostics.Verify(
+                // error CS0041: Unexpected error writing debug information -- 'Insufficient memory to continue the execution of the program.'
```

Comment:
I'm confused. This test is linked to [https://github.com/dotnet/roslyn/issues/75237](https://github.com/dotnet/roslyn/issues/75237) but exhibits the same/current behavior described in the issue.

Is this a different scenario than what customers had reported? Should we report a more specific error here (which limit was hit)?

Reply:
The purpose of these two tests (`_Over` and `_Under`) is to verify the limit is indeed 0x10000. To do that, they use a custom writer which intercepts metadata writes and passes a custom buffer with the specified size. The specific error reporting logic happens before that, so it's not hit. I don't think there's a straightforward way to test that with exact buffer sizes. But `NativeWriterLimit_EndToEnd` tests the specific error (albeit the buffer size cannot be set exactly).

Reply:
Thanks for the context. Makes sense. Consider leaving comments (this is not so much about testing the compiler as testing the limit of library that we depend on)